src/test-support/test-support.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { BlankNode, Literal, NamedNode, Quad, Term } from "rdf-js";
+import { DataFactory } from "n3";
+import { SolidDataset } from "../interfaces";
+
+/**
+ * Expects the specified dataset to have the specified number of matching
+ * quads. If successful, and only one match, then return that quad - otherwise
+ * we return 'null' (meaning 'there were multiple different matching quads for
+ * this pattern).
+ *
+ * Note: we only put the 'graph' component as the last param 'cos we don't yet
+ * assert much on quads in this codebase...
+ *
+ * @param dataset the RDF dataset to match against
+ * @param subject the subject to match (or null to represent a 'wildcard')
+ * @param predicate the predicate to match (or null to represent a 'wildcard')
+ * @param object the object to match (or null to represent a 'wildcard')
+ * @param expected the number of matches expected
+ * @param graph the graph to match (or null to represent a 'wildcard')
+ * @return if only one matching quad found return that quad, else null
+ */
+export function expectMatch(
+  dataset: SolidDataset,
+  subject: string | NamedNode | BlankNode | null,
+  predicate: string | NamedNode | null,
+  object: string | NamedNode | BlankNode | Literal | null = null,
+  expected: number = 1,
+  graph: NamedNode | null = null
+): Quad | null {
+  const rdfSubject = handleSubject(subject);
+  const rdfPredicate = handlePredicate(predicate);
+  const rdfObject = handleObject(object);
+  const rdfGraph = graph === null ? null : graph;
+
+  const matches = dataset.match(rdfSubject, rdfPredicate, rdfObject, rdfGraph);
+
+  if (matches.size !== expected) {
+    throw new Error(
```

Comment:
Yeah, I did consider all that when going through this - but decided to try and keep the impact on the existing tests as minimal as possible initially. And so I catered for all the **_current_** test scenarios with just a single function. That meant the function did need to perform multiple things - i.e., blowing up on expectation failure (i.e., an incorrect match count), returning the single match iff there was just one match.
Now that we know this single function satisfies all the current use-cases across the entire test suite, we can revisit it in a subsequent PR to consider how to split it out into `getMatchCount()` and `returnFirstMatch()` functions, and then use native Jest matchers. So I've added this comment to a 'Tech Debt' ticket to refactor this.

Reply:
I'd prefer to have this one done now though, as otherwise, IMHO this change makes the tests _harder_ to maintain - `expect.assertions` will no longer work, and `eslint-plugin-jest` and the Inrupt linting rules, once applied, can no longer warn us about potential problems. That is in addition to the fact that the same function has different orthogonal behaviours/responsibilities, and the reliance on positional arguments as mentioned elsewhere making it harder to read. Since this is a cleanup PR, it should actually make things cleaner :)

Reply:
Chatted with Jack on this - we're both rather push this clean up to the Hardening sprint to allow this get PR get in sooner rather than later.

Reply:
But it makes my work harder rather than easier, so getting this in now would provide negative value IMHO. I can do the work too (should be super straightforward, TypeScript will tell you exactly where you'd need to update the code), but can't do that right away.

Reply:
We should immediately follow up with a PR to clean this up to prevent this PR for lingering too long. Overall this PR does clean up far more things than this particular issue causes, and we can schedule time to update this next sprint.

Reply:
Cleaning this up is a fraction of the effort that's been spent on it so far - I'll do it at the start of next week, so it won't be lingering too long (for whatever "too long" is).